modules/core/src/test/scala/com/azavea/stac4s/CatalogSpec.scala

```diff
@@ -300,101 +300,114 @@ class CatalogSpec extends AnyFunSpec with Matchers {
             roles = Nil,
-            _type = `image/tiff`.some
+            _type = `image/tiff`.some,
+            extensionFields = ().asJsonObject
             // "eo:bands": [4],
           ),
           "B6" -> StacItemAsset(
             href = "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_B6.TIF",
             title = "SWIR (B6)".some,
             description = "SWIR Band Top Of the Atmosphere".some,
             roles = Nil,
-            _type = `image/tiff`.some
+            _type = `image/tiff`.some,
+            extensionFields = ().asJsonObject
             // "eo:bands": [5],
           ),
           "B7" -> StacItemAsset(
             href = "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_B7.TIF",
             title = "SWIR Band (B7)".some,
             description = "SWIR Band Top Of the Atmosphere".some,
             roles = Nil,
-            _type = `image/tiff`.some
+            _type = `image/tiff`.some,
+            extensionFields = ().asJsonObject
             // "eo:bands": [6],
           ),
           "B8" -> StacItemAsset(
             href = "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_B8.TIF",
             title = "Panchromatic Band (B8)".some,
             description = "Panchromatic Band Top Of the Atmosphere".some,
             roles = Nil,
-            _type = `image/tiff`.some
+            _type = `image/tiff`.some,
+            extensionFields = ().asJsonObject
             // "eo:bands": [7],
           ),
           "B9" -> StacItemAsset(
             href = "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_B9.TIF",
             title = "Cirrus Band (B9)".some,
             description = "Cirrus Band Top Of the Atmosphere".some,
             roles = Nil,
-            _type = `image/tiff`.some
+            _type = `image/tiff`.some,
+            extensionFields = ().asJsonObject
             // "eo:bands": [8],
           ),
           "B10" -> StacItemAsset(
             href = "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_B10.TIF",
             title = "LWIR Band (B10)".some,
             description = "LWIR Band Top Of the Atmosphere".some,
             roles = Nil,
-            _type = `image/tiff`.some
+            _type = `image/tiff`.some,
+            extensionFields = ().asJsonObject
             // "eo:bands": [9],
           ),
           "B11" -> StacItemAsset(
             href = "http://landsat-pds.s3.amazonaws.com/L8/153/025/LC81530252014153LGN00/LC81530252014153LGN00_B11.TIF",
             title = "LWIR Band (B11)".some,
             description = "LWIR Band Top Of the Atmosphere".some,
             roles = Nil,
-            _type = `image/tiff`.some
+            _type = `image/tiff`.some,
+            extensionFields = ().asJsonObject
```

Comment:
I think it makes more sense to provide a default argument than it does to make it optional, especially since the methods for the typeclasses return `ValidatedNel`s -- I'll provide a default and delete all these

Reply:
aa8a265